### PR TITLE
Auth

### PR DESCRIPTION
**Pull request by sikiritskaya**

### CODE REVIEW

File: src/Post/postsController.js [hunk 1]
```diff
@@ -1,10 +1,21 @@
+
 import postsService from "./postsService.js"
 
 class PostsController {
+    async getAll(req, res) {
```
Comment (alexeykurbako) [inline src/Post/postsController.js]: use then().catch() instead of async/await and try/catch

File: src/Post/postsController.js [hunk 2]
```diff
@@ -1,10 +1,21 @@
...
+        try {
+            const posts = await postsService.getAll()
+            return res.json(posts)
+        }
+        catch (e) {
+            res.status(500).json(e)
+        }
+    }
     async create(req, res) {
+       // const {title, body} = req.body
```
Comment (alexeykurbako) [inline src/Post/postsController.js]: Delete comments and console logs please if you don't need them

File: src/Post/postsService.js [hunk 3]
```diff
@@ -1,8 +1,14 @@
+import AuthUser from "../User/Auth/AuthUser.js"
 import Post from "./Post.js"
 
 class PostsService {
+    async getAll() {
```
Comment (alexeykurbako) [inline src/Post/postsService.js]: Could you use then everywhere please instead of async/await?

Comment (alexeykurbako) [reply src/Post/postsService.js]: The same for try-catch

File: src/User/Auth/authController.js [hunk 4]
```diff
@@ -0,0 +1,77 @@
...
+            console.log(e)
+        }
+    }
+    async getAllPosts(req,res){
+        try{
+            const posts = await authService.getAllPosts(req.params.id)
+            return res.json(posts)
+        }
+        catch(e){
+            console.log(e)
```
Comment (alexeykurbako) [inline src/User/Auth/authController.js]: Send message to the front-end please

File: src/User/Auth/authController.js [hunk 5]
```diff
@@ -0,0 +1,77 @@
...
+        }
+    }
+    async getAllUsers(req, res){
+        try{
+           const users = await authService.getAllUsers()
+           return res.json(users)
+        }
+        catch(e){
+            console.log(e)
```
Comment (alexeykurbako) [inline src/User/Auth/authController.js]: Send message to the front-end please

File: src/User/Auth/authService/authService.js [hunk 6]
```diff
@@ -0,0 +1,66 @@
...
+            await mailService.sendActivationMail(username, email, confirmationCode)
+            return user
+        }
+        catch (e) {
+            console.log(e)
+        }
+
+    }
+    async activate(confirmationCode) {
+        const user = await AuthUser.findOne({ confirmationCode })
```
Comment (alexeykurbako) [inline src/User/Auth/authService/authService.js]: Seems we need to add linter to check code styles. There are no ; in the end of string

File: src/User/Auth/authService/authService.js [hunk 7]
```diff
@@ -0,0 +1,66 @@
...
+                throw new Error ('Pending Account. Please verify your email.')
+            } 
+            return user
+        }
+        catch (e) {
+            console.log(e)
+        }
+    }
+    async getAllPosts(id) {
+        const posts = await Post.find({userId: id})
```
Comment (alexeykurbako) [inline src/User/Auth/authService/authService.js]: unused variable

File: src/User/Auth/authService/authService.js [hunk 8]
```diff
@@ -0,0 +1,66 @@
...
+        return posts
+        //const user = await AuthUser.findOne({ _id: id }).populate('posts')
+        //return user.posts
+    }
+    async getAllUsers() {
+        const users = await AuthUser.find()
```
Comment (alexeykurbako) [inline src/User/Auth/authService/authService.js]: unused variable

File: src/User/Auth/sendEmail.js [hunk 9]
```diff
@@ -1,3 +1,6 @@
+//это было до, логику поместила в mailService
```
Comment (alexeykurbako) [inline src/User/Auth/sendEmail.js]: Why do we need this file?

File: src/Post/routerPosts.js [hunk 10]
```diff
@@ -4,8 +4,9 @@ import postsController from "./postsController.js";
 
 const postsRouter = new Router();
 
-postsRouter.post('/posts', postsController.create)
-postsRouter.delete('/posts/:id', postsController.delete)
+postsRouter.get('/posts', postsController.getAll)
+postsRouter.post('/post/', postsController.create)
```
Comment (alexeykurbako) [inline src/Post/routerPosts.js]: /posts instead of /post/

File: src/User/Auth/authRouter.js [hunk 11]
```diff
@@ -0,0 +1,17 @@
...
+
+const authRouter = new Router();
+
+authRouter.post('/registration', [
+    check('username', 'should not be empty').notEmpty(),
+    check('password', 'should has 4-10 symbols').isLength({min:4, max:10})
+], authController.registration)
+authRouter.post('/login', authController.login) 
+authRouter.get('/activate/:link', authController.activate)
+authRouter.get('/test', authController.getAllUsers )
```
Comment (alexeykurbako) [inline src/User/Auth/authRouter.js]: test?

File: src/User/Auth/authRouter.js [hunk 12]
```diff
@@ -0,0 +1,17 @@
+import { Router } from "express";
+import authController from "./authController.js";
+import { check } from 'express-validator';
+
+const authRouter = new Router();
+
+authRouter.post('/registration', [
+    check('username', 'should not be empty').notEmpty(),
+    check('password', 'should has 4-10 symbols').isLength({min:4, max:10})
+], authController.registration)
```
Comment (alexeykurbako) [inline src/User/Auth/authRouter.js]: sign-up instead of registration

File: src/User/Auth/authRouter.js [hunk 13]
```diff
@@ -0,0 +1,17 @@
...
+authRouter.post('/login', authController.login)
```
Comment (alexeykurbako) [inline src/User/Auth/authRouter.js]: sign-in instead of login

File: src/User/Auth/authRouter.js [hunk 14]
```diff
@@ -0,0 +1,17 @@
...
+
+authRouter.post('/registration', [
+    check('username', 'should not be empty').notEmpty(),
+    check('password', 'should has 4-10 symbols').isLength({min:4, max:10})
+], authController.registration)
+authRouter.post('/login', authController.login) 
+authRouter.get('/activate/:link', authController.activate)
+authRouter.get('/test', authController.getAllUsers )
+authRouter.get('/users/:id', authController.getAllPosts)
+authRouter.post('/users/create', authController.create)
```
Comment (alexeykurbako) [inline src/User/Auth/authRouter.js]: /users instead of /users/create

File: src/User/Auth/authService/authService.js [hunk 15]
```diff
@@ -0,0 +1,66 @@
...
+import Post from "../../../Post/Post.js";
+
+class AuthService {
+    async registration(username, password, email) {
+        const newUser = await AuthUser.findOne({ username })
+        const newEmail = await AuthUser.findOne({ email })
+        if (newUser || newEmail) {
+            throw new Error('such user exists')
+        }
+        const hashPassword = await bcrypt.hash(password, 5)
```
Comment (alexeykurbako) [inline src/User/Auth/authService/authService.js]: The next step will be adding salt and usage sha256 algorithm. For now leave it as it

File: src/Post/postsController.js (hunk 2 above)
Comment (sikiritskaya) [reply src/Post/postsController.js]: deleted

File: src/User/Auth/authController.js (hunk 4 above)
Comment (sikiritskaya) [reply src/User/Auth/authController.js]: sent

File: src/User/Auth/authController.js (hunk 5 above)
Comment (sikiritskaya) [reply src/User/Auth/authController.js]: sent

File: src/Post/routerPosts.js (hunk 10 above)
Comment (sikiritskaya) [reply src/Post/routerPosts.js]: fixed

File: src/User/Auth/authRouter.js (hunk 11 above)
Comment (sikiritskaya) [reply src/User/Auth/authRouter.js]: fixed

File: src/User/Auth/authRouter.js (hunk 12 above)
Comment (sikiritskaya) [reply src/User/Auth/authRouter.js]: fixed

File: src/User/Auth/authRouter.js (hunk 13 above)
Comment (sikiritskaya) [reply src/User/Auth/authRouter.js]: fixed

File: src/Post/postsController.js (hunk 1 above)
Comment (sikiritskaya) [reply src/Post/postsController.js]: used then()

File: src/Post/postsService.js (hunk 3 above)
Comment (sikiritskaya) [reply src/Post/postsService.js]: used then()

File: src/User/Auth/authService/authService.js (hunk 6 above)
Comment (sikiritskaya) [reply src/User/Auth/authService/authService.js]: added linter

File: src/User/Auth/authService/authService.js (hunk 7 above)
Comment (sikiritskaya) [reply src/User/Auth/authService/authService.js]: deleted

File: src/User/Auth/sendEmail.js (hunk 9 above)
Comment (sikiritskaya) [reply src/User/Auth/sendEmail.js]: deleted

File: src/User/Auth/authRouter.js (hunk 14 above)
Comment (sikiritskaya) [reply src/User/Auth/authRouter.js]: renamed

**Pull request closed without merging**